### ISSUE TITLE
setup/teardownTimeout behavior when set to 0s

### ISSUE THREAD

**Issue by cuonglm** [grafana/k6#1174]
Currently, setup/teaddownTimeout duration if set to `0s` means we can't run setup/teardown functions anymore. With script:

```
import http from 'k6/http';
import { k6 } from 'k6';

export function setup() {
  sleep(1);
}

export default function() {
  const response = http.get("http://test.loadimpact.com");
};
```

Run:

```
$ K6_SETUP_TIMEOUT=0s k6 run t.js

          /\      |‾‾|  /‾‾/  /‾/
     /\  /  \     |  |_/  /  / /
    /  \/    \    |      |  /  ‾‾\
   /          \   |  |‾\  \ | (_) |
  / __________ \  |__|  \__\ \___/ .io

  execution: local
     output: -
     script: t.js

    duration: -,  iterations: 1
         vus: 1, max: 1

ERRO[0000] Setup timeout                                 error="setup: Timeout during setup"
ERRO[0000] Setup timeout
exit 100
```

In most libraries, setting timeout duration to `0` often means disable the timeout. Should we adopt that behavior to `k6` as well?

**Comment by cuonglm** [comment]
We do have `minIterationDuration` which will be disable when set to 0.

**Comment by JustJordanT** [comment]
Is this something that still need to be worked on?

**Comment by mstoykov** [comment]
Hu @JustJordanT - yes it appears nobody has worked on it and it still a problem.

**Comment by tsukasaI** [comment]
Hi, can I take this one?

**Comment by tsukasaI** [comment]
Hi, I opened [PR](https://github.com/grafana/k6/pull/3898).
I would appreciate your reply. Thank you!

**Comment by mstoykov** [comment]
I don't think we should have a way to disable the timeout in more or less the same vein as #2701 - there is no need for that.

The current proposed PR disables the setup/teardown execution, but I am not particularly certain that is a good idea either. There already is an option for that, and I will expect that ... yes most people will expect this to mean it is infinite, and then it just doesn't run.

I am kind of for this just being a validation error similar to if you define it as negative number.


cc @grafana/k6-core

**Comment by tsukasaI** [comment]
@mstoykov 

Thank you for comment.

Oh Yeah, I would think we should have a validation for that this property is positive number.

Can I introduce the validation?

**Comment by tsukasaI** [comment]
Hi @mstoykov 
I updated my PR. #3898 

I would appreciate your review. Thank you!

**Comment by mstoykov** [comment]
This was fixed with #3898 thanks to @tsukasaI 🙇
